src/lib/blkalloc/fixed_blk_allocator.cpp

```diff
@@ -55,27 +54,54 @@ blk_num_t FixedBlkAllocator::init_portion(BlkAllocPortion& portion, blk_num_t st
 bool FixedBlkAllocator::is_blk_alloced(BlkId const& b, bool use_lock) const { return true; }
 
 BlkAllocStatus FixedBlkAllocator::alloc([[maybe_unused]] blk_count_t nblks, blk_alloc_hints const&, BlkId& out_blkid) {
-    HS_DBG_ASSERT_EQ(nblks, 1, "FixedBlkAllocator does not support multiple blk allocation yet");
-    return alloc_contiguous(r_cast< BlkId& >(out_blkid));
-}
+    if (m_state == state_t::RECOVERING) {
```

Comment:
nit: could we move this to init_portion during recovery ?

Reply:
Problem is there is no way to know when the recovery is completed, that is why we are doing on the first alloc.